R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+#' c("c1" = "color1", "c2" = "color2")
+#'
+#' @inheritParams coh_coverage
+#' @param doses_count_color color assigned to the doses count
+#' @param coverage_color color assigned to the coverage calculation
+#' @param cumulative if TRUE returns the cumulative number of doses
+#' over the time window
+#' @return 2-axis ggplot2 plot of vaccine coverage and daily doses
+#' @examples
+#' data("cohortdata")
+#' start_cohort <- as.Date("2044-01-01")
+#' end_cohort <- as.Date("2044-12-31")
+#' date_interval <- c(start_cohort, end_cohort)
+#' plot_coverage(
+#'   data = cohortdata,
+#'   vacc_date_col = "vaccine_date_1",
+#'   unit = "month",
+#'   doses_count_color = "steelblue",
+#'   coverage_color = "mediumpurple",
+#'   date_interval = date_interval,
+#'   cumulative = FALSE
+#' )
+#' @export
+plot_coverage <- function(data,
+                          vacc_date_col,
+                          unit = c("day", "month", "year"),
+                          doses_count_color = "steelblue",
+                          coverage_color = "mediumpurple",
+                          date_interval = FALSE,
+                          cumulative = FALSE) {
+  checkmate::assert_data_frame(
+    data,
+    min.rows = 1L
+  )
+  checkmate::assert_character(
+    vacc_date_col,
+    min.len = 1L
+  )
+  checkmate::assert_names(
+    colnames(data),
+    must.include = c(vacc_date_col)
+  )
+  checkmate::assert_date(
+    data[[vacc_date_col]]
+  )
+  unit <- match.arg(unit, several.ok = FALSE)
+  checkmate::assert_string(
+    unit
+  )
+  checkmate::assert_logical(
+    cumulative, len = 1
+  )
+  checkmate::assert_character(
+    c(doses_count_color, coverage_color)
+  )
+
+  coverage <- coh_coverage(data = data,
+    vacc_date_col = vacc_date_col,
+    unit = unit,
+    date_interval = date_interval
+  )
+  if (cumulative) {
+    coverage$dose_plot <- coverage$cum_dose
+  } else {
+    coverage$dose_plot <- coverage$dose
+  }
+
+  colors <- c("c1" = doses_count_color, "c2" = coverage_color)
+  lbls <- c("c1" = "Doses", "c2" = "Coverage")
+  plt <- ggplot2::ggplot() +
+    ggplot2::geom_bar(ggplot2::aes(x = coverage$date,
+        y = coverage$dose_plot,
+        fill = "c1"
+      ), stat = "identity", alpha = 0.6
+    ) +
+    ggplot2::geom_line(ggplot2::aes(x = coverage$date,
+        y = coverage$coverage * max(coverage$dose_plot),
+        color = "c2", group = 1
+      ), linewidth = 1.3, linetype = 2
+    ) +
+    ggplot2::xlab("") +
+    ggplot2::ylab(paste0("Doses per ", unit)) +
+    ggplot2::scale_y_continuous(
+      labels = scales::label_number(scale_cut = scales::cut_short_scale()),
+      sec.axis = ggplot2::sec_axis(~. / max(coverage$dose_plot),
+        labels = scales::percent,
+        name = "Percentage of coverage"
+      )
+    ) +
+    ggplot2::scale_fill_manual(name = "",
+      values = colors,
+      labels = lbls
+    ) +
+    ggplot2::scale_color_manual(name = "",
+      values = colors,
+      labels = lbls
+    ) +
```

Comment:
This chunk could be simplified a bit to take advantage of how ggplot treats the `colour` and `fill` arguments inside `aes()`.

I would also suggest specifying the X-axis scale as a `scale_date()` and providing the format. For data where the units are in years, you might want a better format such as simply "%Y", which could be added with a small `ifelse(unit == "year", "%Y", "%b %Y")`.

I would also suggest using `geom_col()` rather than `geom_bar()` as that's more correct when you pass both the X and Y values. It's also not necessary to specify a `stat`.

```suggestion
 plt <- ggplot2::ggplot(coverage) +
    ggplot2::geom_col(
      ggplot2::aes(
        x = .data$date,
        y = .data$dose_plot,
        fill = "Doses"
      ),
      alpha = 0.6
    ) +
    ggplot2::geom_line(
      ggplot2::aes(
        x = .data$date,
        y = .data$coverage * max(.data$dose_plot),
        group = 1,
        colour = "Coverage"
      ),
      linetype = "dashed"
    ) +
    ggplot2::scale_x_date(name = NULL, date_labels = "%b %Y") +
    ggplot2::scale_y_continuous(
      name = paste0("Doses per ", unit),
      labels = scales::label_number(
        scale_cut = scales::cut_short_scale()
      ),
      sec.axis = ggplot2::sec_axis(~. / max(coverage$dose_plot),
        labels = scales::percent,
        name = "Percentage of coverage"
      )
    ) +
    ggplot2::scale_fill_manual(
      name = NULL, values = doses_count_color
    ) +
    ggplot2::scale_color_manual(
      name = NULL, values = coverage_color
    ) +
```

Reply:
Done. See https://github.com/epiverse-trace/vaccineff/pull/31/commits/cbaf081d7d7f0a2792685563bb7e17893d3fe91f
The group argument can be safely removed, apparently, I kept it by mistake from a previous test. 
Probably in the next PR, I'll transform the second axis plot into a barplot. It seems that is the standard way to plot it in various papers.